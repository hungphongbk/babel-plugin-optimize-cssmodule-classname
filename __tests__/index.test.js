import path from "path";
import webpack from "webpack";
import MemoryFs from "memory-fs";

function compile(rules) {
  const compiler = webpack({
    mode: "development",
    context: __dirname,
    entry: "./fixtures/App.js",
    output: {
      path: __dirname,
      filename: "bundle.js",
    },
    module: { rules },
  });

  compiler.outputFileSystem = new MemoryFs();

  return new Promise((resolve, reject) => {
    compiler.run((err, stats) => {
      if (err) reject(err);

      resolve(stats.toJson().modules.map((m) => m.source));
    });
  });
}
