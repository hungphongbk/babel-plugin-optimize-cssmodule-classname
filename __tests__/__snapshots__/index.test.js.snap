// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack loader should convert file 1`] = `
Array [
  "// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../node_modules/css-loader/dist/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".App_App__3btAC {\\\\n  font-size: 1rem;\\\\n}\\\\n.App_HelloWorld__1SR0T {\\\\n  display: inline-block;\\\\n  margin: 0;\\\\n  padding: 0;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"App\\": \\"App_App__3btAC\\",
	\\"HelloWorld\\": \\"App_HelloWorld__1SR0T\\"
};
module.exports = exports;
",
  "import React from \\"react\\";
import styles from \\"./App.module.css\\";

function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: styles.App
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    className: styles.HelloWorld
  }, \\"Hello World\\"));
}

export default App;",
  "var api = require(\\"!../../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js\\");
            var content = require(\\"!!../../node_modules/css-loader/dist/cjs.js??ref--4-oneOf-2-1!../../node_modules/postcss-loader/src/index.js??postcss!./App.module.css\\");

            content = content.__esModule ? content.default : content;

            if (typeof content === 'string') {
              content = [[module.id, content, '']];
            }

var options = {};

options.insert = \\"head\\";
options.singleton = false;

var update = api(content, options);

var exported = content.locals ? content.locals : {};



module.exports = exported;",
]
`;
